### ISSUE TITLE
Redesign of a Crop tool?

### ISSUE THREAD

**Issue by maboleth** [Beep6581/RawTherapee#5825]
I find Crop tool not that much intuitive and clunky.

Key issues:
- Cropped parts remain at all times and seriously alter perception of how the image looks like, even with "fit crop to screen".
- Moving the remaining field around the image requires two hands - shift pressed.
- Thumbnail gets considerably smaller with large black rectangle around the image - harder to see, depending on the crop.

I think there should be an option for a RT to remove the rest of the image from the screen once you set your crop - it's unneeded and in case of the thumbnails, alters the usability. One simple icon saying "Your image is cropped and not full size" would do. You could, of course, go back to the crop tool and re-set everything again if needed.

And second, moving the remaining field with a mouse should also be default, without the shift key.

As it is now, Crop in RT remains clunky. It is a tool that is probably heavily used by any photographer, thus should be as intuitive and responsive as possible.

**Comment by heckflosse** [comment]
> Cropped parts remain at all times

![grafik](https://user-images.githubusercontent.com/13733487/86404627-43718300-bcb0-11ea-893f-0105ff8a4881.png)

**Comment by maboleth** [comment]
Thanks for that. If you didn't tell me this, I'd never figure this out. Seriously. I searched in the Preferences for this and since I didn't find it, with "Fit to crop" icon there, I was like 100% sure this was a feature of RT.
Of course it is said in the manual - but who reads the manual for the crop tool unless something big/advanced changed? :D


Also, what about the thumbnails? Thumbnails still show the cropped areas. The bigger the crop, the less useful/visible the thumbnail.

**Comment by maboleth** [comment]
Well, if you move the image on the screen, cropped areas still pop-in. I really am not sure I like this.

I still think it would be logical for cropped areas to be 100% invisible unless being _in_ the crop tool.
If I cropped the image, why would I still be reminded it's cropped when I move it or see it in the thumbnail all the time?

It's almost like RT is telling me:
'_but you cropped your image!!!!'_
'yeah I did...so?'
_'Seee!!!!!!! Are you sure?!? You can still crop it again with the crop tool and the hand tool if you change your mind! I will make it pop-in like a flashback!!'
'And oh, I will also make sure you see it in the thumbnail too!!'
'You will forever be reminded that you. cropped. your. image. !! '_ 

xD

**Comment by Thanatomanic** [comment]
There are a few related issues to this. #5076 #5678 #4083 #4263
I also remember an issue about having the automatic WB work only on the cropped selection, but I can't find it.

> As it is now, Crop in RT remains clunky. It is a tool that is probably heavily used by any photographer, thus should be as intuitive and responsive as possible.

I agree with this. Though, the code that handles the preview and crop is ~~a hot mess~~ complicated and it will break _a ton_ of other stuff if we start modifying it. Bad code, maybe, a lot of work to right the wrongs, absolutely...

**Comment by maboleth** [comment]
Thanks for letting me know that.
That's a bit unfortunate, considering how much work it requires. However, better sooner than never I guess? 

Not sure of your priorities as of now, Crop is important tool nonetheless. But even those slight QoL changes, like middle-click pan would help. And I believe that's not hard to code?

I'm not a programmer, but I could test it to the extreme if you guys need any workforce for that. I did some beta testing before.

> There are a few related issues to this. #5076 #5678 #4083 #4263
> I also remember an issue about having the automatic WB work only on the cropped selection, but I can't find it.

I tested this on several of my RAW files and couldn't reproduce it. Auto WB always work on the entire image, no matter how cropped it is.

**Comment by Thanatomanic** [comment]
Thanks for wanting to test. This may be something to revisit when we start thinking about RT6.0. It's always good to discuss issues from a usability perspective.

> I tested this on several of my RAW files and couldn't reproduce it. Auto WB always work on the entire image, no matter how cropped it is.

That was my point: currently it takes the whole image, but I think it makes much more sense to only calculate it based on the selected crop. However, there are situations where this approach would fail. So the optimal solution would probably be to be able to select which area you want to use for auto WB. But that's another issue entirely #5717.

**Comment by maboleth** [comment]
@Thanatomanic 
Is it realistic to expect middle mouse button press for panning the cropped selection in 5.9? Or resolving bugs #5076 & #4083?

The first guy requested it in 2017, three years ago. And it shouldn't be that difficult to change I guess? That would be some nice and easy QoL change.

**Comment by Thanatomanic** [comment]
@maboleth The fact that these issues (not bugs per se, they are marked as enhancements) are out for such a long time means either one of three things:
1. It _is_ difficult to change
2. No active developer thinks it is a particularly enjoyable thing to investigate (which is unfortunately often true for QoL stuff)
3. No active developer thinks it is actually worthwhile to change the current behavior (which is all up to personal taste)

I'm a number 4 atm, I have too little time to investigate. 🙃

**Comment by maboleth** [comment]
Fair enough. But if the key bind/change needs 3+ years (and at least three separate reports), what are the chances for something bigger? :-/

I understand you guys are doing this in your free time. And I'm grateful on what you already did for the open source photography world. Make no mistake on that. It's just that comment like that leave a little room for us to 'discuss issues from a usability perspective.'

**Comment by Thanatomanic** [comment]
I know, I wish it were different. Just as a general commentary on the state of things: if you look at our colleagues over at darktable, they have managed to gain enormous momentum in their development because they have a few people that practically (or literally) have made it their job. If you look at the commit list for RawTherapee in the last year, you see that all changes are done by only a handful of people. That is the unfortunate and slow reality.

**Comment by ror6ax** [comment]
You guys rock! RT is an amazing piece of software!

**Comment by heckflosse** [comment]
> But if the key bind/change needs 3+ years (and at least three separate reports), what are the chances for something bigger? :-/

Well, Capture Sharpening, raw CA correction iterations, Pixelshift, etc... were a lot bigger...

**Comment by BorisBrock** [comment]
I just migrated from CaptureOne to RawTherapee. So far RT rocks :-)

The only thing I stumbled upon when getting started was the crop tool. It is implemented in a very weird way, which is (in my opinion) not very intuitive. These two points mentioned in the initial post are also what bothers me most:

- Cropped parts remain at all times and seriously alter perception of how the image looks like, even with "fit crop to screen".
- Moving the remaining field around the image requires two hands - shift pressed.

Since the crop tool is the first thing I use on nearly all my images, I would love to see it being improved. Also for new users (like I am/was), the current state of the tool is very confusing.

Thank you all for creating this wonderful tool. Keep up the good work!

**Comment by adam4** [comment]
this is halfway between crop and resize, but I've been using RT to prepare files for printing, and I think an option to add a bleed of certain size would be useful

eg, I use the ISO A size ratio, but it isn't exactly right for A size + bleed

it's almost ok, for A5 + 3mm bleed on all sides I resize to 2551x1818, then RT corrects the height to 1804, so the image is almost the right size, I didn't have problems with the printed files yet, but it does mean an  incorrectly sized bleed in Y axis

**Comment by Lawrence37** [comment]
@adam4 There is #6791 for a crop bleed feature.

**Comment by adam4** [comment]
thanks, I'll redirect myself there :)
